processing.go

```
@@ -0,0 +1,885 @@
+				e.Post(Event{
+					Type:         DataAcceptance,
+					Callsign:     msg.Identifier,
+					ToController: msg.TrackOwner,
+				})
+			} else {
+				if trk := comp.TrackInformation[msg.Identifier]; trk != nil {
+					comp.TrackInformation[msg.Identifier] = &TrackInformation{
+						TrackOwner:        msg.TrackOwner,
+						HandoffController: msg.HandoffController,
+						FlightPlan:        trk.FlightPlan,
+					}
+					delete(comp.ContainedPlans, msg.BCN)
+					e.Post(Event{
+						Type:         DataAcceptance,
+						Callsign:     msg.Identifier,
+						ToController: msg.TrackOwner,
+					})
+				} else { // send an IF msg
+					e.Post(Event{
+						Type:         DataRejection,
+						Callsign:     msg.Identifier,
+						ToController: msg.TrackOwner,
+					})
+				}
+
+			}
+
+		case AcceptRecallTransfer:
+			// When we send an accept message, we set the track ownership to us.
+			// when we receive an accept message, we change the track ownership to the receiving controller.
+			// When we send a recall message, we tell our system to stop the flashing.
+			// When we receive a recall message, we keep the plan and if we click the track, it is no longer able to be accepted
+			// We can infer whether its a recall/ accept by the track ownership that gets sent back.
+
+			info := comp.TrackInformation[msg.Identifier]
+			if info == nil {
+				break
+			}
+
+			if msg.TrackOwner != comp.TrackInformation[msg.Identifier].TrackOwner { // has to be an accept message. (We initiated the handoff here)
+				if entry, ok := comp.TrackInformation[msg.Identifier]; ok {
+					entry.TrackOwner = msg.TrackOwner
+					entry.HandoffController = ""
+					comp.TrackInformation[msg.Identifier] = entry
+				}
+			} else { // has to be a recall message. (we received the handoff)
+				delete(comp.TrackInformation, msg.Identifier)
+			}
+		}
+	}
+	clear(comp.RecievedMessages)
+}
+
+// For NAS codes
+func (comp *ERAMComputer) CreateSquawk() Squawk {
+	for {
```

Infinite `for` loops like this make me nervous since a) they may never terminate, in which case vice would just hang and b) in this case, when there are few squawk codes available, the loop may run many times, with each one looping over every flight plan and every `TrackInformation` to make sure it's ok.

As an alternative, you could have `ERAMComputer` maintain a `map[int]interface{}` of available squawk codes. At initialization time you can seed that with everything except `0o1200` and the other invalid squawk codes. Then if you maintain the invariant that if a squawk code is in use, it is not in that map, you just need to pick a random squawk code out of the map and delete it from the map here.

Extra bonus, since go randomizes map iteration, you could just do something like:
```
for sq := range comp.availableSquawkCodes {
  delete(comp.availableSquawkCodes, sq)
  return sq
}
```

This also would simplify error handling; you could return an error of the map was empty.